runtime/src/iree/task/post_batch.c

```diff
@@ -68,19 +72,25 @@ iree_host_size_t iree_task_post_batch_select_worker(
   // worker's queue to finish. Note that we only consider workers idle if we
   // ourselves in this batch haven't already queued work for them (as then they
   // aren't going to be idle).
-  iree_task_affinity_set_t worker_idle_mask =
-      iree_atomic_task_affinity_set_load(
-          &post_batch->executor->worker_idle_mask, iree_memory_order_relaxed);
-  worker_idle_mask &= ~post_batch->worker_pending_mask;
-  iree_task_affinity_set_t idle_affinity_set = affinity_set & worker_idle_mask;
-  if (idle_affinity_set) {
-    return iree_task_post_batch_select_random_worker(post_batch,
-                                                     idle_affinity_set);
+  iree_task_affinity_set_t nonpending_affinity_set =
+      affinity_set & ~post_batch->worker_pending_mask;
+  iree_host_size_t selected_worker;
```

Comment:
style: always default initialize (helps debugging, if anything, as the value is never garbage)